finagle/h2/src/main/scala/com/twitter/finagle/buoyant/h2/Message.scala

```diff
@@ -60,7 +61,8 @@ object Headers {
     def toSeq = synchronized(current)
     override def toString = s"""Headers(${toSeq.mkString(", ")})"""
     def contains(key: String) = toSeq.exists { case (k, _) => key == k }
-    def get(key: String) = toSeq.collect { case (k, v) if key == k => v }
+    def get(key: String) = toSeq.collectFirst { case (k, v) if key == k => v }
```

Comment:
If this is a super hot method that we're trying to optimize, you could try getting rid of the partial function allocation and just iterate over the seq and build the result.